src/main.cpp

```diff
@@ -1134,6 +1133,13 @@ bool ContextualCheckTransaction(
                 REJECT_INVALID, "bad-tx-too-many-orchard-actions");
         }
 
+        if (orchard_bundle.GetNumActions() > 0 && !orchard_bundle.OutputsEnabled() && !orchard_bundle.SpendsEnabled()) {
+            return state.DoS(
+                dosLevelPotentiallyRelaxing,
+                error("ContextualCheckTransaction(): Orchard actions are present, but flags do not permit Orchard spends or outputs"),
+                REJECT_INVALID, "bad-tx-orchard-flags-disable-actions");
+        }
```

Comment:
This should not be a contextual check.

Reply:
The restrictions on `vShieldedSpend.size()`, `vShieldedOutput.size()`, and `GetNumActions()` also don't need to be contextual (the former two are weaker than what is already implied by the 2 MB block size limit).

Reply:
Also, the check at https://github.com/zcash/zcash/blob/5802be342781b7b7d37f61b70f333fbb14ab5409/src/main.cpp#L1510-L1514 is made redundant by this check (and therefore is no longer in the spec).